search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/ext/GenerativeQAParameters.java

```diff
@@ -0,0 +1,114 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.ext;
+
+import com.google.common.base.Preconditions;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ObjectParser;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Defines parameters for generative QA search pipelines.
+ *
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+public class GenerativeQAParameters implements Writeable, ToXContentObject {
+
+    private static final ObjectParser<GenerativeQAParameters, Void> PARSER;
+
+    private static final ParseField CONVERSATION_ID = new ParseField("conversation_id");
+    private static final ParseField LLM_MODEL = new ParseField("llm_model");
+    private static final ParseField LLM_QUESTION = new ParseField("llm_question");
+
+    static {
+        PARSER = new ObjectParser<>("generative_qa_parameters", GenerativeQAParameters::new);
+        PARSER.declareString(GenerativeQAParameters::setConversationId, CONVERSATION_ID);
+        PARSER.declareString(GenerativeQAParameters::setLlmModel, LLM_MODEL);
+        PARSER.declareString(GenerativeQAParameters::setLlmQuestion, LLM_QUESTION);
+    }
+
+    @Setter
+    @Getter
+    private String conversationId;
+
+    @Setter
+    @Getter
+    private String llmModel;
+
+    @Setter
+    @Getter
+    private String llmQuestion;
+
+    public GenerativeQAParameters(StreamInput input) throws IOException {
+        this.conversationId = input.readOptionalString();
+        this.llmModel = input.readOptionalString();
+        this.llmQuestion = input.readString();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder xContentBuilder, Params params) throws IOException {
+        return xContentBuilder.field(CONVERSATION_ID.getPreferredName(), this.conversationId)
+            .field(LLM_MODEL.getPreferredName(), this.llmModel)
+            .field(LLM_QUESTION.getPreferredName(), this.llmQuestion);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (conversationId != null) {
```

Comment:
Just an optimization, `writeOptionalString` already does a null check. 
We can remove it here.